apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue

```diff
@@ -113,58 +111,76 @@ const { widgetState, widgetFrameProps, widgetFrameEventHandlers } = useWidget(pr
+        const refinedFields = getWidgetTableDateFields(widgetState.granularity, widgetState.dateRange, _textOptions, 'value_sum');
 
         // set width of table fields
         const groupByFieldWidth = refinedFields.length > 4 ? '28%' : '34%';
         const otherFieldWidth = refinedFields.length > 4 ? '18%' : '22%';
 
-        const fixedFields = [{
+        const groupByField: Field = {
             label: USAGE_TYPE_VALUE_KEY,
             name: USAGE_TYPE_VALUE_KEY,
             width: groupByFieldWidth,
-        }];
-        if (state.fieldsKey === 'usage_quantity_sum') {
-            fixedFields.push({
-                label: 'Usage Unit',
-                name: 'usage_unit',
-                width: otherFieldWidth,
-            });
-        }
+        };
         return [
-            ...fixedFields,
+            groupByField,
             ...refinedFields.map((field) => ({ ...field, width: otherFieldWidth })),
         ];
     }),
-    tableData: computed<WidgetTableData[]>(() => {
-        const tableData = sortArrayInObjectArray(
-            getRefinedDateTableData(state.data?.results, widgetState.dateRange, ['cost_sum', 'usage_quantity_sum']),
+    tableData: computed<Data[]>(() => {
+        const tableData: Data[] = sortArrayInObjectArray(
+            getRefinedDateTableData(state.data?.results, widgetState.dateRange, ['value_sum']),
             DATE_FIELD_NAME,
-            ['cost_sum', 'usage_quantity_sum'],
-        );
+            ['value_sum'],
+        ).map((data) => {
+            let value = data[USAGE_TYPE_VALUE_KEY] ?? 'Unknown';
+            if (state.fieldsKey === 'usage_quantity') value = `${value}${data.usage_unit ? ` (${data.usage_unit})` : ''}`;
+            return {
+                ...data,
+                [USAGE_TYPE_VALUE_KEY]: value,
+            };
+        });
+
         return tableData;
     }),
-    legends: computed<Legend[]>(() => getXYChartLegends(state.data?.results, USAGE_TYPE_VALUE_KEY)),
+    legends: computed<Legend[]>(() => {
+        const data = state.data?.results ?? [];
+        const legends: Legend[] = getXYChartLegends(data, USAGE_TYPE_VALUE_KEY)
+            .map((l, i) => {
+                let label = l.label;
+                if (state.fieldsKey === 'usage_quantity') label = `${label}${data[i]?.usage_unit ? ` (${data[i]?.usage_unit})` : ''}`;
+                return { ...l, label };
+            });
+        return legends;
+    }),
     showLegendsOnTable: computed(() => widgetState.options.legend_options?.enabled && widgetState.options.legend_options.show_at === 'table'),
     chart: null as XYChart | null,
+    showChart: computed(() => {
+        if (state.fieldsKey === 'cost') return true;
+        if (!state.data?.results) return true;
+        // hide chart when there are different usage_unit in data
+        return uniqBy(state.data.results, 'usage_unit').length === 1;
+    }),
```

Comment:
note: chart must be hidden if there are different value of usage_unit in each array data.